react-docs/src/pages/Actions.tsx

```diff
@@ -0,0 +1,252 @@
+import React from 'react';
+import CodePreview from './CodePreview';
+import { MainHeading } from './atoms';
+
+const Introduction = () => {
+  return (
+    <div>
+      <MainHeading>Actions</MainHeading>
```

Comment:
I probably miss some context, but why not use markdown for the text? I guess you have considered another option: By having all the documentation as `.md` files would make it easier to write and I would argue it scales better. The snippets could lie in the same folder as the text.

Reply:
Yeah, I see your point! For now it is written in JSX because it possible to use the CodePreview component, and I believe all of us should be capable of reading and updating JSX easily. 

However as we discussed with @tlaundal, in a v2 of this we could probably move to markdown and find a way to "enhance" them with code previews. I think this is how [antd](https://raw.githubusercontent.com/ant-design/ant-design/master/components/alert/demo/error-boundary.md) produce their documentation. I first tried to see if we could re-use their approach/code, but it wasn't trivial.